apps/condo/domains/property/components/panels/Builder/forms/UnitForm.tsx

```diff
@@ -95,41 +95,35 @@ const UnitForm: React.FC<IPropertyMapModalForm> = ({ builder, refresh }) => {
 
     const isUnitUnique = useMemo(() => {
         let isUnitLabelUnique = true
+        const selectedUnit = builder.getSelectedUnit()
         if (mode === MapEditMode.AddUnit) {
-            isUnitLabelUnique = builder.validateUniqueUnitLabel()
+            isUnitLabelUnique = builder.validateInputUnitLabel(selectedUnit, label)
+            setDuplicatedUnitIds(builder.duplicatedUnits)
         } else if (mode === MapEditMode.EditUnit) {
-            const selectedUnit = builder.getSelectedUnit()
             if (!selectedUnit) {
                 return false
             }
-
-            const unitPlacementChanged = selectedUnit.floor !== floor
-                || selectedUnit.section !== section
-                || unitType !== selectedUnit.unitType
-            const labelChanged = selectedUnit.label !== label
-            const labelValidation = labelChanged
-                ? builder.validateUniqueUnitLabel(label, 'section')
-                : false
-
-            isUnitLabelUnique = unitPlacementChanged
-                ? (!labelChanged || labelValidation)
-                : labelValidation
+            isUnitLabelUnique = builder.validateInputUnitLabel(selectedUnit, label)
+            setDuplicatedUnitIds(builder.duplicatedUnits)
```

Comment:
Maybe just add special class to all of the found units with native javascript api? That's not trigger rerender cycle at all

Reply:
So, on the contrary, a rerender cycle is needed, otherwise the information in the UnitButton will simply not be relevant

Reply:
I'm not sure it's a good idea to do this bypassing react logic. At a minimum, this breaks the code consistency and looks like a hack.